src/framework/components/rigid-body/system.js

```diff
@@ -86,14 +88,16 @@ class SingleContactResult {
  * @property {Vec3} point The point on the entity where the contact occurred, in world space.
  * @property {Vec3} pointOther The point on the other entity where the contact occurred, in world space.
  * @property {Vec3} normal The normal vector of the contact on the other entity, in world space.
+ * @property {number} impulse The total accumulated impulse applied by the constraint solver during the last substep. Describes how hard two objects collide.
```

Comment:
This seems like quite an abstract concept. If I'm reading this value, how would I interpret it to do something useful with it. I mean, what kind of range can I expect? What are the units?

Reply:
Within a single simulation step, Bullet engine accumulates the impulse over the substeps and then writes that total value into contact point of the collision manifold. The value will depend on the object mass and the speed it is moving, so should be tried against actual application to find the suitable values.

Here is an example with a 1 kg ball:
https://playcanvas.com/editor/scene/1269287

A 10 kg ball, for example, would produce the same values 10 times larger.

Reply:
In one of the projects, I am using this to spawn some dust and sound effects, when a vehicle lands too hard on a surface. I did have to test some values and smash the car a few times to find a suitable value for it. I'm afraid you won't be able to pick a single value that is usable across different objects, unless they share the same physical properties.

Reply:
OK, fair enough. I mean, it's fine - I was just curious.